lisa/microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -1338,6 +1426,707 @@ def _set_block_size_env_var(self, testcase: str) -> None:
+        # Only reset if currently pinned (non-zero value)
+        result = self.node.execute(
+            "for xps in /sys/class/net/*/queues/tx-*/xps_cpus; do "
+            '[ -f "$xps" ] && current=$(cat $xps) && '
+            '[ "$current" != "00000000" ] && [ "$current" != "0" ] && '
+            "echo 0 | sudo tee $xps || true; done",
+            shell=True,
+            sudo=True,
+        )
+
+        # Enforce MTU (default 1500, configurable via perf_mtu)
+        expected_mtu = self.perf_mtu
+
+        # Find primary NIC
+        result = self.node.execute(
+            "ip route get 1.1.1.1",
+            shell=True,
+        )
+
+        if result.exit_code == 0 and result.stdout.strip():
+            # Extract device name using Python regex
+            # Expected format: "1.1.1.1 via ... dev eth0 ..."
+            match = re.search(r"dev\s+(\S+)", result.stdout)
+            if not match:
+                self._log.debug(
+                    f"Could not parse primary NIC from: {result.stdout.strip()}"
+                )
+                return
+
+            primary_nic = match.group(1)
+
+            # Get current MTU
+            current_mtu_result = self.node.execute(
+                f"ip link show {primary_nic}",
+                shell=True,
+            )
+
+            if current_mtu_result.exit_code == 0:
+                # Extract MTU using Python regex
+                # Expected format: "... mtu 1500 ..."
+                mtu_match = re.search(r"mtu\s+(\d+)", current_mtu_result.stdout)
+                if not mtu_match:
+                    self._log.debug(
+                        f"Could not parse MTU from: {current_mtu_result.stdout.strip()}"
+                    )
+                    return
+
+                current_mtu = int(mtu_match.group(1))
+
+                if current_mtu != expected_mtu:
+                    # Set MTU
+                    self.node.execute(
+                        f"sudo ip link set {primary_nic} mtu {expected_mtu}",
+                        shell=True,
+                        sudo=True,
+                    )
+
+    def _log_thermal_health(self, stage: str) -> None:
+        """
+        Log thermal health: sensors, throttle counts, frequency state.
+
+        Args:
+            stage: Execution stage (preflight, anchor_failure, suite_end)
+        """
+        self._log.info(f"=== Thermal Health [{stage}] ===")
+
+        # sensors output (if available)
+        result = self.node.execute(
+            "sensors | head -20",
+            shell=True,
+        )
+        if result.exit_code == 0 and result.stdout.strip():
+            for line in result.stdout.split("\n")[:10]:
+                if line.strip():
+                    self._log.info(f"  {line}")
+        else:
+            self._log.info("  Sensors: not available")
+
+        # Throttle counts
+        result = self.node.execute(
+            "for f in /sys/devices/system/cpu/cpu*/thermal_throttle/"
```

Comment:
Is this thermal throttle data even present in virtualized environments like ours? Does it even have useful data?

Reply:
its useful for baremetal, it will have : (this is just logging data for any debugging later)
core_throttle_count - How many times this CPU core reduced frequency due to temperature
package_throttle_count - How many times the entire CPU package (socket) throttled
core_throttle_total_time_ms - Total milliseconds spent in throttled state